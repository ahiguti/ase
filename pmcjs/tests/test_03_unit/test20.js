
/* tied array + method */

Perl.Use("TieTest");
var fs = Perl.Functions["TieTest"];
var ms = Perl.Methods["TieTest"];
var x = ms.newat(0);
x[3] = 5;
x[1] = 6;
x[9] = 7;
//x[11] = 7;
print("x1=" + x[1]);
print("enum1");
for (var i in x) {
  print("ITER k=" + i + " v=" + x[i]);
}
delete x[9];
print("enum2");
for (var i in x) {
  print("ITER k=" + i + " v=" + x[i]);
}
print("enum3");
for (var i = 0; i < x.length; ++i) {
  print("ROOP i=" + i + " v=" + x[i]);
}
var r = x.foo(15);
print(r);

