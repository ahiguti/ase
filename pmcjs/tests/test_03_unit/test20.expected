foo invoked
x1=6
enum1
ITER k=0 v=undefined
ITER k=1 v=6
ITER k=2 v=undefined
ITER k=3 v=5
ITER k=4 v=undefined
ITER k=5 v=undefined
ITER k=6 v=undefined
ITER k=7 v=undefined
ITER k=8 v=undefined
ITER k=9 v=7
enum2
ITER k=0 v=undefined
ITER k=1 v=6
ITER k=2 v=undefined
ITER k=3 v=5
ITER k=4 v=undefined
ITER k=5 v=undefined
ITER k=6 v=undefined
ITER k=7 v=undefined
ITER k=8 v=undefined
enum3
ROOP i=0 v=undefined
ROOP i=1 v=6
ROOP i=2 v=undefined
ROOP i=3 v=5
ROOP i=4 v=undefined
ROOP i=5 v=undefined
ROOP i=6 v=undefined
ROOP i=7 v=undefined
ROOP i=8 v=undefined
30
